Makefile

```diff
@@ -0,0 +1,8 @@
+unit_tests:
```

Comment:
Whats this?

Reply:
This is to allow us easily run unit & integration tests locally on demand. `make unit_tests` to quickly run all unit tests. On linux there is built-in tab completion for Makefiles. So you type make [tab] and get suggestions. I also see this could be extended in future with `make all_tests`, `make docs`, etc.
But I can drop it if you feel this should be moved away from this PR